src/main/java/org/jabref/model/pdf/search/SearchResult.java

```diff
@@ -15,37 +18,48 @@
 import org.apache.lucene.search.highlight.SimpleHTMLFormatter;
 import org.apache.lucene.search.highlight.TextFragment;
 
+import static org.jabref.model.pdf.search.SearchFieldConstants.ANNOTATIONS;
 import static org.jabref.model.pdf.search.SearchFieldConstants.CONTENT;
 import static org.jabref.model.pdf.search.SearchFieldConstants.MODIFIED;
+import static org.jabref.model.pdf.search.SearchFieldConstants.PAGE_NUMBER;
 import static org.jabref.model.pdf.search.SearchFieldConstants.PATH;
 
 public final class SearchResult {
 
     private final String path;
-    private final String content;
+
+    private final int pageNumber;
     private final long modified;
 
     private final float luceneScore;
-    private String html;
+    private List<String> contentResultStringsHtml;
+    private List<String> annotationsResultStringsHtml;
 
     public SearchResult(IndexSearcher searcher, Query query, ScoreDoc scoreDoc) throws IOException {
         this.path = getFieldContents(searcher, scoreDoc, PATH);
-        this.content = getFieldContents(searcher, scoreDoc, CONTENT);
+        this.pageNumber = Integer.parseInt(getFieldContents(searcher, scoreDoc, PAGE_NUMBER));
         this.modified = Long.parseLong(getFieldContents(searcher, scoreDoc, MODIFIED));
         this.luceneScore = scoreDoc.score;
 
-        TokenStream stream = new EnglishStemAnalyzer().tokenStream(CONTENT, content);
+        String content = getFieldContents(searcher, scoreDoc, CONTENT);
+        String annotations = getFieldContents(searcher, scoreDoc, ANNOTATIONS);
+
+        Highlighter highlighter = new Highlighter(new SimpleHTMLFormatter("<b>", "</b>"), new QueryScorer(query));
 
-        Highlighter highlighter = new Highlighter(new SimpleHTMLFormatter(), new QueryScorer(query));
+        TokenStream contentStream = new EnglishStemAnalyzer().tokenStream(CONTENT, content);
```

Comment:
Wrap the TokenStream in a try with resources ( ) block, it implements AutoCloseable , otherwise you will leak the stream 
https://lucene.apache.org/core/7_3_0/core/org/apache/lucene/analysis/TokenStream.html